src/better-shifting.cow

```diff
@@ -422,10 +422,191 @@ or by a scalar shift amount which applies to all elements in the \tcode{simd::ve
 
 The following changes are relative to \ref[N5008].
 
-\Btodo{
-This.
+\Bnote{
+The \tcode{shl} and \tcode{shr} functions should immediately precede the
+\tcode{rotl} and \tcode{rotr} functions in the wording. Therefore, if this paper
+and \ref[P3764] are adopted in the same meeting, the \tcode{shl} and \tcode{shr}
+functions should be inserted between the \tcode{msb_to_mask} and \tcode{rotl}
+functions.
 }
 
+\h3{[bit]}
+
+In \eelis{bit.syn}, change the synopsis as follows:
+
+\Bdiff{
+\itemdecl{
+namespace std {
+  \html{[\N{HORIZONTAL ELLIPSIS}]}
+
+\ins{  // \serif{[bit.shift], shifting}
+  template<class T>
+    constexpr T shl(T x, int s) noexcept;
+  template<class T>
+    constexpr T shr(T x, int s) noexcept;
+}
+
+  // \serif{[bit.rotate], rotating}
+  template<class T>
+    constexpr T rotl(T x, int s) noexcept;
+  template<class T>
+    constexpr T rotr(T x, int s) noexcept;
+
+  \html{[\N{HORIZONTAL ELLIPSIS}]}
+}
+}
+}
+
+In \eelis{bit}, add a new subclause immediately preceding \eelis{bit.rotate}:
+
+\Bins{
+\html-h3{Shifting \stable-ref{bit.shift}}
+
+\itemdecl{
+template<class T>
+  constexpr T shl(T x, int s) noexcept;
+}
+
+\itemdescr{
+\constraints
+\tcode{T} is a signed or unsigned integer type\iref{basic.fundamental}.
+
+\effects
+If \tcode{s} is negative, the behavior is erroneous and an
+implementation-defined value is returned. Otherwise, returns the value of
+\tcode{x} multiplied by 2\sup{\tcode{s}}.
+\wg21_note{If \tcode{s} is greater than or equal to the bit width of \tcode{T},
```

Comment:
```suggestion
\br\wg21_note{If \tcode{s} is greater than or equal to the bit width of \tcode{T},
```
`\wg21_note` is not a block, and it looks better when it's on a separate line imo.